searx/search.py

```diff
@@ -53,6 +48,49 @@
         exit(1)
 
 
+class EngineRef:
+
+    __slots__ = 'name', 'category', 'from_bang'
+
+    def __init__(self, name: str, category: str, from_bang: bool=False):
+        self.name = name
+        self.category = category
+        self.from_bang = from_bang
+
+    def __str__(self):
+        return "(" + self.name + "," + self.category + "," + str(self.from_bang) + ")"
+
+
+class SearchQuery:
```

Comment:
What is the benefit of moving `SearchQuery` from `query.py`?

Reply:
The purpose is to have one simple object to run a query: the searx.search.Search object.
It takes a SearchQuery object as input and returns a ResultContainer object.
SearchQuery contains only simple Python types, it doesn't depend on the web / front-end.

The front-end parses the user inputs using webadapter.py.

And webadapter.py uses:
* query.py to parse the query & bangs.
* preferences.py to parse the preferences from the cookies.

With SearchQuery in search.py, it creates a clean split between front-end and back-end.